server/interpreter/data/object.go

```diff
@@ -16,11 +16,48 @@
 
 package data
 
-// Object represents typical JavaScript objects with (optional)
-// prototype, properties, etc.
-type Object struct {
+// Object represents any JavaScript value (primitive, object, etc.).
+type Object interface {
+	/// Any Object is a valid Value.
+	Value
+
+	// Proto returns the prototype (parent) object for this object.
+	// N.B. this is object.__proto__, not Constructor.prototype!
+	Proto() Object
+
+	// Get returns the current value of the given property or an
+	// ErrorMsg if that was not possible.
+	Get(name string) (Value, *ErrorMsg)
```

Comment:
My suggestion would be one of the following:

- If ErrorMsg is really only intended to be used where JavaScript would throw an exception, rename the type to Exception.

- Return an ordinary error interface and add another interface that can extract an exception from an error. The caller would typecast to that interface to check if the error should throw an exception (`err.(Exception)`).